src/csl/mod.rs

```diff
@@ -1470,6 +1475,11 @@ pub struct CitationItem<'a, T: EntryLike> {
     initial_idx: usize,
 }
 
+impl<'a, T: EntryLike> Hash for CitationItem<'a, T> {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        self.entry.key().hash(state);
+    }
+}
```

Comment:
Why did you decide to factor out this Hash impl?

Reply:
Actually, I don't remember for now. I remember that I used HashSet somewhere and needed to change hash's behaviour. I think that I'll try to remove this and return deriving back.